### PR TITLE
Fix vanity command by taking the network into account

### PR DESCRIPTION
**Pull request by bkchr**
Fixes: https://github.com/paritytech/substrate/issues/7177

### CODE REVIEW

File: client/cli/src/commands/vanity.rs [hunk 1]
```diff
@@ -180,6 +187,17 @@ mod tests {
...
 
+	#[test]
+	fn generate_key_respects_network_override() {
+		let seed = generate_key::<sr25519::Pair>("ab", Ss58AddressFormat::PolkadotAccount).unwrap();
+		assert!(
+			sr25519::Pair::from_seed_slice(&hex::decode(&seed[2..]).unwrap())
+				.unwrap()
+				.public()
+				.to_ss58check_with_version(Ss58AddressFormat::PolkadotAccount)
+				.contains("ab"));
```
Comment (kianenigma) [inline client/cli/src/commands/vanity.rs:198]:
```suggestion
				.contains("ab")
		);
```

File: client/cli/src/commands/vanity.rs [hunk 2]
```diff
@@ -180,6 +187,17 @@ mod tests {
 				.contains("j"));
```
Comment (kianenigma) [inline client/cli/src/commands/vanity.rs:187]:
```suggestion
				.contains("j")
		);
```

**Review by kianenigma — APPROVED** [review]

File: client/cli/src/commands/vanity.rs [hunk 3]
```diff
@@ -171,7 +178,7 @@ mod tests {
 
 	#[test]
 	fn test_generation_with_single_char() {
-		let seed = generate_key::<sr25519::Pair>("j").unwrap();
+		let seed = generate_key::<sr25519::Pair>("j", Default::default()).unwrap();
```
[the remark this answers is not in the document]
Comment (bkchr) [reply client/cli/src/commands/vanity.rs]: I also changed it to `ab`. In the end it isn't such a great test. But this test is random and we don't want to wait until the end of the universe for a result :D

**Comment by bkchr** [comment]
bot merge

**Comment by bkchr** [comment]
bot merge

**Comment by bkchr** [comment]
bot merge

**Pull request merged**